src/requests/requests.controller.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpException,
+  HttpStatus,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+} from '@nestjs/common'
+import { Permissions } from 'src/auth/casl-ability.factory'
+import { AuthorizationSubject } from 'src/auth/decorator/authorizationSubject.decorator'
+import { CurrentUser } from 'src/auth/decorator/current-user.decorator'
+import { JwtAuth } from 'src/auth/decorator/jwtAuth.decorator'
+import { RequiredPermission } from 'src/auth/decorator/requiredPermission'
+import { UserEntity } from 'src/users/dto/UserEntity.dto'
+import { CreateRequestDto } from './dto/CreateRequest.dto'
+import { RequestDetailsDto } from './dto/RequestDetails.dto'
+import { RequestPreviewDto } from './dto/RequestPreview.dto'
+import { UpdateRequestDto } from './dto/UpdateRequest.dto'
+import { RequestsService } from './requests.service'
+
+@JwtAuth()
+@Controller('requests')
+@AuthorizationSubject('ConsultationRequest')
+export class RequestsController {
+  constructor(private readonly requestsService: RequestsService) {}
+
+  @Get()
+  async findAll(): Promise<RequestPreviewDto[]> {
+    return this.requestsService.findAll()
+  }
+
+  @Get(':id')
+  async findOne(
+    @Param('id', ParseIntPipe) id: number,
+  ): Promise<RequestDetailsDto> {
+    try {
+      return this.requestsService.findOne(id)
+    } catch {
+      throw new HttpException(
+        'A konzi kérés nem található!',
+        HttpStatus.NOT_FOUND,
+      )
+    }
+  }
+
+  @Post()
+  async create(
+    @Body() createRequest: CreateRequestDto,
+    @CurrentUser() user: UserEntity,
+  ) {
+    return this.requestsService.create(createRequest, user)
+  }
```

Comment:
the service throws an exception if the foreign key constraint is violated on the subjectId field. That exception should be caught, and a 400 thrown isntead